src/plugins/discover/public/components/common/error_callout.tsx

```diff
@@ -16,48 +23,73 @@ import { useDiscoverServices } from '../../hooks/use_discover_services';
 interface Props {
   title: string;
   error: Error;
+  isEsqlMode?: boolean;
 }
 
-export const ErrorCallout = ({ title, error }: Props) => {
-  const { core } = useDiscoverServices();
+export const ErrorCallout = ({ title, error, isEsqlMode }: Props) => {
+  const { core, docLinks } = useDiscoverServices();
   const { euiTheme } = useEuiTheme();
 
   const searchErrorDisplay = renderSearchError(error);
 
   return (
     <EuiEmptyPrompt
-      iconType="error"
-      color="danger"
+      icon={<EuiIcon size="l" type="error" color="danger" />}
```

Comment:
We also have another similar prompt for incomplete results case which we might want to update at some point too.

<img width="1172" alt="Screenshot 2024-09-02 at 11 19 13" src="https://github.com/user-attachments/assets/f56fc1b9-5b0a-40fc-8d9f-e94de3d411b2">

To reproduce, create a data view for `kibana_sample_data_flights,kibana_sample_data_logs` with `timestamp` as a time field. change time range to get 0 results. Add a custom QueryDSL filter:

```
{
    "error_query": {
        "indices": [
          {
            "name": "kibana_sample_data_logs",
            "shard_ids": [0, 1],
            "error_type": "exception",
            "message": "Testing shard failures!"
          }
        ]
    }
}
```

Reply:
@ryankeairns do you want to do this on this iteration?

Reply:
🤔 this one seems to have a bit of a different feel to it. The phrasing 'This cluster...' sounds more like a backend, system issue than what this PR covers which is largely a user-oriented issue with how they constructed their query of filter. I will convert this comment to an issue to track separately.